### DISCUSSION TITLE
Multi trigger flow

### DISCUSSION THREAD

**Discussion by NigmaX** [directus/directus#15536] [Feature Requests]
So, I have a suggestion to improve the "flows" structure a bit.

I create multiple flows for the same purpose, only with different triggers.
It would be nice to have one of the following.

 - support multiple triggers in a flow (manual + event)
 - create grouping abilities in the flows setting screen (so we can group it in an accordion by categories)
 - attach a group flow (mentioned in the previous point) to the different collections (that way, it will include all the flows that are within it)

Let me know your thoughts about it.
Is it doable?
Many thanks for considering my request.

**Comment by br41nslug** [comment]
Actually having support for multiple triggers like mixing `schedule` and an `event` might be unintuitive however i do like the mixing with manual triggers.
I'm thinking having a debug trigger always available where you can manually provide a JSON payload to test things could be valuable for admins. The event + manual combination is a good one tho perhaps the event triggers should have an option to also manually trigger on the collection it's watching events for 🤔

**Comment by colemahatmccreary** [comment]
Not sure how recommended patterns will change as the available triggers/operations grow, but a clean solution might be:

Keep one flow (Flow A) responsible for executing the main process on the event/payload. Then create any number of additional flows (Flows B, C, etc) – each with their own triggers based on requirements. 

When triggered, flows B, C, etc. are responsible for constructing a 'Flow A' payload and passing it down via the 'Trigger Flow' operation.

**Reply by rijkvanzanten** [reply]
This is my recommendation as well ☝🏻

**Comment by rijkvanzanten** [comment]
I'd second @colemahatmccreary's recommendation above. Having multiple triggers for the same flow means that you can no longer rely on the contents of `$trigger` at all.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
